### PR TITLE
fix and refactor swap data tasks implementation and call from restart…

### PR DESCRIPTION
**Pull request by francois-vincent**
…_kraken

### CODE REVIEW

File: fabfile/component/kraken.py [hunk 1]
```diff
@@ -181,7 +187,8 @@ def restart_kraken(instance, test=True, wait=True):
     """
     instance = get_real_instance(instance)
     wait = get_bool_from_cli(wait)
-    swap_data_nav(instance)
+    if env.host == env.roledefs['eng'][0].split('@')[1]:
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: this does seem strange.

why do you do this ?

File: fabfile/component/kraken.py [hunk 2]
```diff
@@ -157,21 +157,27 @@ def test_all_krakens(wait=False):
...
     temp_target = instance.temp_target_lz4_file
     plain_target = instance.plain_target_lz4_file
-    if files.is_file(temp_target) and files.is_file(plain_target) and \
-       files.getmtime(temp_target) > files.getmtime(plain_target):
-        swap_temp = os.path.join(os.path.dirname(temp_target), 'x')
-        files.move(plain_target, swap_temp)
-        files.move(temp_target, plain_target)
-        files.move(swap_temp, temp_target)
+    with warn_only():
+        if files.is_file(temp_target) and files.is_file(plain_target) and \
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: isn't there a problem when plain_target does not exists ?

Comment (francois-vincent) [inline fabfile/component/kraken.py]: that's why I say "with warn_only" !

Comment (antoine-de) [inline fabfile/component/kraken.py]: I don't think it's a question of the warn_only, but you want to move the `temp_target` even in there is no `plain_target` no ?

Comment (francois-vincent) [inline fabfile/component/kraken.py]: you're right. This case will happen when an instance is deployed the first time

File: fabfile/component/kraken.py [hunk 3]
```diff
@@ -181,7 +190,8 @@ def restart_kraken(instance, test=True, wait=True):
     """
     instance = get_real_instance(instance)
     wait = get_bool_from_cli(wait)
-    swap_data_nav(instance)
+    if env.host_string == env.roledefs['eng'][0]:
```
Comment (antoine-de) [inline fabfile/component/kraken.py]: I don't understand this (well I think I do, but it does not seems straightforward)

isn't it better to add this on a taks only for 'tyr_master' ?

Comment (francois-vincent) [inline fabfile/component/kraken.py]: pour le coup, non.
On est déjà dans une tâche avec rôles, du coup on aurait un produit cartésien et on veut une réduction.

Comment (pbougue) [inline fabfile/component/kraken.py]: I don't comprend what you mean :)

Comment (antoine-de) [inline fabfile/component/kraken.py]: and can't it be called beforehand ? I would really prefer to minimize the number of tricks on env.roledefs

**Comment by francois-vincent** [comment]
done

**Pull request merged**